packages/@aws-cdk/aws-appmesh/lib/mesh.ts

```diff
@@ -0,0 +1,368 @@
+  /**
+   * The backend services this node expects to send traffic to
+   *
+   * @default none
+   */
+  readonly backends?: VirtualNodeBackendProps[];
+  /**
+   * @default none
+   * if not specified must call addListeners(), addPortMappings() or addPortAndHealthCheckMappings()
+   */
+  readonly listeners?: ListenerProps;
+}
+
+/**
+ * A utility enum created for the egressFilter type property, the default of DROP_ALL,
+ * allows traffic only to other resources inside the mesh, or API calls to amazon resources.
+ *
+ * @default Drop_All
+ */
+export enum MeshFilterType {
+  /**
+   * Allows all outbound traffic
+   */
+  Allow_All = 'ALLOW_ALL',
+  /**
+   * Allows traffic only to other resources inside the mesh, or API calls to amazon resources
+   */
+  Drop_All = 'DROP_ALL',
+}
+
+/**
+ * Used to populate the spec property for the mesh, only available option today is egressFilter
+ *
+ */
+export interface MeshSpec {
+  /**
+   * Egress filter to be applied to the Mesh
+   *
+   * @default DROP_ALL
+   */
+  readonly egressFilter?: MeshFilterType;
+}
+
+/**
+ * Used for importing and exporting Mesh(es)
+ */
+export interface MeshImportProps {
+  /**
+   * The AppMesh name to import
+   */
+  readonly meshName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  readonly meshArn?: string;
+  /**
+   * The unique identifier for the mesh
+   */
+  readonly meshUid: string;
+}
+
+/**
+ * Interface wich all Mesh based classes MUST implement
+ */
+export interface IMesh extends cdk.IConstruct {
+  /**
+   * The name of the AppMesh mesh
+   */
+  readonly meshName: string;
+  /**
+   * The Amazon Resource Name (ARN) of the AppMesh mesh
+   */
+  readonly meshArn?: string;
+  /**
+   * The unique identifier for the mesh
+   */
+  readonly meshUid: string;
+  /**
+   * Adds a VirtualRouter to the Mesh with the given id and props
+   *
+   * @param id the name for the virtual router
+   * @param props set of VirtualRouterBaseProps to apply to the virtual router
+   */
+  addVirtualRouter(id: string, props: VirtualRouterBaseProps): VirtualRouter;
+  /**
+   * Adds a VirtualService with the given id
+   *
+   * @param id the name for the given service
+   * @param props a set of VirtualServiceBaseProps to create the service
+   */
+  addVirtualService(id: string, props: VirtualServiceBaseProps): VirtualService;
+  /**
+   * Adds a VirtualNode to the Mesh
+   *
+   * @param id the name for the VirtualNode
+   * @param props a set of AddVirtualNodeProps
+   */
+  addVirtualNode(id: string, props: AddVirtualNodeProps): VirtualNode;
+  /**
+   * Exports the Mesh properties to re-use in other stacks
+   */
+  export(): MeshImportProps;
+}
+
+/**
+ * Represents a new or imported AppMesh mesh
+ *
+ * @example cdk.Construct
+ * @implements IMesh
+ */
+export abstract class MeshBase extends cdk.Construct implements IMesh {
+  /**
+   * The name of the AppMesh mesh
+   *
+   * @abstract
+   * @type {string}
+   * @memberof MeshBase
```

Comment:
The `@type`, `@memberof` and `@abstract` annotations are not necessary in `TypeScript`, please remove those.

Reply:
will be committed soon with the rest